feat: Improve the styling

cindyli opened the pull request:
This PR improves the styling but doesn't complete all works. Further css and backend event handlers are required to get the UI to fully function.

In the new design, the content on the left side menu could be a city list or search results. As these 2 contents will not appear at the same time, the markup in this PR assumes these contents will be rendered using the same container `.fld-mapviz-city-and-results`. @amb26, please let me know if you'd like to use different containers for 2 lists.

Some TODOs from my notes:

1. SVGs that are embedded as background images need to be rewritten as inline SVGs. Only inline SVGs are able to respond to the UIO contrast change. The new SVGs in this PR are embedded as inline SVGs;
2. Add click event handlers for these buttons:
   * The filter control button `.fld-filter-control` for opening/closing the a11y feature list panel. 
   * The arrow up/down button `.fl-mapviz-city-and-results-expand-button` for opening/closing the city list/search results panel. 
   Note that set proper aria-label and aria-expanded properly according to the opening/closing state.
3. When features in the a11y feature list are selected, an event handler is required to increase the number of selected in the filter control box.
4. CSS for the `body` element needs be move out of the map CSS file. Otherwise, when integrating the map into the WeCount website, it will interfere with the website `body` styling. We probably would like to consider to move demo specific css into a separate file. This file can be excluded when building the minified & concatenated css for exports.
5. Checkboxes on the a11y feature list are not tab focussable.

amb26 commented:
Hi @cindyli - Thanks for these improvements:

In response to the first question, yes, we should maintain separate containers for the city list and search results, as well as separate selectors for all of their contents. They only have the superficial similarity that they contains lists of things, and we will want to style and bind to them separately.

Next - I like the approach of embedding the SVG icons within the file, since it solves the preloading issue as well as the styling issue. However, i) we should produce some kind of minimal build system which produces an HTML template from the contents of the "img" directory so we don't need to maintain this cut and paste by hand, ii) it seems problematic to use this system with the leaflet marker/icon system https://leafletjs.com/examples/custom-icons/ which expects the images to be resolvable as images - it is hardwired to render these as <img src=" tags.

Finally, there are some layout and responsiveness problems with the new styling. I include the original layout, the updated layout, and the behaviour of the updated layout when making a narrow window to test responsiveness.

![orig-css](https://user-images.githubusercontent.com/561418/103357550-5ae8c780-4aab-11eb-8fd8-780a58677d41.jpg)
![cindy-css-annotated](https://user-images.githubusercontent.com/561418/103357556-5e7c4e80-4aab-11eb-87ab-0cf68eea0c6e.jpg)
![cindy-responsive-css](https://user-images.githubusercontent.com/561418/103357561-61773f00-4aab-11eb-9405-473cf36311c9.jpg)

i) The alignment of the filter button strips with the top and bottom of the map has been lost. Compare with the original designs at https://www.figma.com/file/0lcLol3X5MmOXackT2YbHJ/WeCount-website - note that in the desktop design, the top of the top filter icon panel is level with the top of the overall page, not with the map, and that the bottom of the Apply Filters/Reset Filters panel is level with the bottom of the map
ii) The responsive design is badly corrupted. On making a narrow window, we lose all of the search results which are pushed off the bottom of the page, a scrollbar appears which scrolls only a height of about 50 pixels, and the map attribution label "Leaflet" jumps off the map onto the bottom of the page.
iii) The alignment of the Apply Filters/Reset Filters panel has degraded. This used to have 40 pixels to the left and 35 to the right which was sort of acceptable, but now only has 30 pixels to the right which is visibly distorted. This should be centred.

amb26 commented:
Looks like we could render the markers inline using this scheme: https://leafletjs.com/reference-1.7.1.html#divicon

amb26 commented on index.html:
```diff
@@ -55,129 +55,168 @@
 
     <body>
-        <h2>Accessibility Map for COVID-19 assessment centres</h2>
+        <h1>Accessibility Map for COVID-19 assessment centres</h2>
```
Mismatched open and close tag here

amb26 commented:
Cheers @cindyli - I've made an updated work in progress pull at https://github.com/inclusive-design/covid-data-monitor/pull/5 - this addresses most of your TODOs from above and contains some cut and paste lifts from this branch, but because of the corrupted responsive layout I haven't merged all the contents in. I've started with inlining a few SVGs but the next steps are the SVG inlining build, the cities view, querying based on postcode and then implementation of actual filtering, for which we'll need to produce a new dataset.

cindyli commented:
@amb26, the responsive design in this branch is still far from complete. But at least the updates so far support all functionality and selector changes you made in #5 and the markup structure should be stable now.

Could you take a look and give a try to merge this branch into #5 so our works can base off the same markup and css, which will significantly reduce conflicts at merging our branches. Thanks.

amb26 commented:
Thanks, @cindyli - could you reissue this as a pull against #5 so I can more clearly review what the changes are? Cheers! Or else perhaps you could merge my #5 .

cindyli commented:
> Thanks, @cindyli - could you reissue this as a pull against #5 so I can more clearly review what the changes are? Cheers! Or else perhaps you could merge my #5 .

@amb26, good idea and done. This branch does have your #5 merged in. I reissued https://github.com/amb26/covid-data-monitor/pull/2 against your #5 branch for an easier review. Thanks.

cindyli committed:
fix: use css to add box-shadow for some icons

cindyli committed:
fix: implement the new design for the control section

cindyli committed:
fix: continues on implememting the responsive design

cindyli committed:
Merge remote-tracking branch 'antranig/design-impl' into improve-css

cindyli committed:
fix: remove pagination related css

cindyli commented:
@amb26, I've merged your #5 into this PR. 

More css work is added to implement the new responsive design. Now there are two buttons, [one for the desktop view](https://github.com/cindyli/covid-data-monitor/blob/improve-css/src/html/template.html#L14-L17) and [the other for the mobile view](https://github.com/cindyli/covid-data-monitor/blob/improve-css/src/html/template.html#L155-L161), for controlling the visibility of the accessibility filter. I was hoping sharing the same css selectors of `fl-mapviz-filter-control` and `fl-mapviz-filter-count` will get these buttons work the same way. However, this triggers an infusion error about "more than one such selector is detected ...". I wonder if you have some insights about whether it's possible to share the same css selectors in this case. Thanks.

amb26 commented:
No need to share the selector, we can just duplicate the component definition and bind it to the other selector.

amb26 commented:
I think we should move to a model where this work is positioned as a pull against my in-progress pull, or else merge my pull request, so that it is more easy to see what work is being done.

cindyli commented:
> I think we should move to a model where this work is positioned as a pull against my in-progress pull, or else merge my pull request, so that it is more easy to see what work is being done.

Good point. I'll merge your pull request.

cindyli committed:
Merge branch 'main' into improve-css

cindyli committed:
feat: improve checkboxes on the a11y filter panel

cindyli committed:
fix: improve custom checkboxes on the mobile view

cindyli committed:
Merge branch 'main' into improve-css

cindyli committed:
feat: add github linting workflow and linted

cindyli committed:
fix: place the template into the proper container in index.html

cindyli committed:
fix: move mobile height control css out of demo.css

cindyli commented:
@amb26, it seems I broke the map in this PR when merging in your #5 pull request: markers are no longer displayed on the map. I haven't figured out what's causing the problem. Could you help to take a look? Thanks.

cindyli committed:
Merge branch 'main' into improve-css

cindyli committed:
fix: resolve conflicts with merging in the main branch

cindyli committed:
fix: improve the city list style

cindyli committed:
fix: more style change to implement the new design for the filter count

cindyli committed:
fix: improve css as well as adding js to update two filter count values

cindyli commented:
@amb26, your #6 pull request has been merged into the main. This PR is: 1) up to date with all your work; 2) implement the latest styling changes; 3) include covidMap.js changes to remove the toggling of the filter panel and update filter count values on both desktop and mobile views.

Could you review this pull request? Once it's merged, we can work off the same code base.

What I'm thinking to do next is: 1. use inline SVGs for all SVGs; 2. update https://github.com/inclusive-design/wecount.inclusivedesign.ca/pull/415 to find out how well this demo works with the existing style of the WeCount website.

Off the top of my head, other TODOs include: 
1. Control the visibility of various panels at different states;
2. The mobile css for the hospital description panel is not in place yet. Waiting on the design for it to be finalized;
3. Add the placeholder text for the search box on the mobile view.

cindyli committed:
fix: revert the implementation of custom checkboxs on the a11y panel

cindyli commented:
@amb26, as discussed, I've reverted the css implementation of the custom checkbox for the a11y filter. Also fixed the elliptical shape of the filter count container. Ready for another look. Thanks.

I will start to look at adding the focus/hover styling for controls.
